internal/cmd/clone/clone.go

```diff
@@ -19,14 +43,32 @@ func (o *options) AddFlags(cmd *cobra.Command) {
 		"",
 		"specify branch to check out",
 	)
+	cmd.Flags().VarP(
+		&o.expectedRootKeys,
+		"root-key-path",
+		"r",
+		"path(s) of root keys to verify when cloning the repo.",
+	)
 }
 
 func (o *options) Run(cmd *cobra.Command, args []string) error {
 	var dir string
 	if len(args) > 1 {
 		dir = args[1]
 	}
-	_, err := repository.Clone(cmd.Context(), args[0], dir, o.branch)
+
+	expectedRootKeys := make([]*tuf.Key, len(o.expectedRootKeys))
+
+	for index, keyPath := range o.expectedRootKeys {
+		key, err := common.LoadPublicKey(keyPath)
+		if err != nil {
+			return err
+		}
+
+		expectedRootKeys[index] = key
```

Comment:
Curious (ie this is fine), but does append() reallocate the underlying array before we actually hit the limit? Is that why you chose to use the index? I figured append would hold off on reallocating given `make` specifies the capacity, but I don't know the exact threshold it uses to reallocate. :smile:

Reply:
Technicaly assigning by index is slightly faster, since append has a small, minute overhead of checking if the array has a high enough capacity to hold a new element, when append does not have to reallocate. Append does not reallocate before the capacity is fully used.

Reply:
I was wondering if there was an optimization for this, but I _think_ not. Admittedly I don't have any real experience writing / reading compiler code so... :smile: